# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/TransceiverImpl.h
# Commit: 09db50c8f370
# Full Hash: 09db50c8f370b69f186f3e6b7ce6154a984b40b4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Pass codec configs by value/reference instead of pointers. r=ng
#   
#   This patch mainly sets the stage for using codec configs with StateMirroring.
#   It also makes the code a bit simpler, and removes the need to handle the nullptr
#   case.
# ==============================================================================

diff -r d1552c620c11 -r 09db50c8f370 dom/media/webrtc/jsapi/TransceiverImpl.h
--- a/dom/media/webrtc/jsapi/TransceiverImpl.h	Wed Aug 11 11:33:21 2021 +0200
+++ b/dom/media/webrtc/jsapi/TransceiverImpl.h	Wed Apr 14 17:36:40 2021 +0200
@@ -135,11 +135,11 @@
 
   static nsresult NegotiatedDetailsToAudioCodecConfigs(
       const JsepTrackNegotiatedDetails& aDetails,
-      std::vector<UniquePtr<AudioCodecConfig>>* aConfigs);
+      std::vector<AudioCodecConfig>* aConfigs);
 
   static nsresult NegotiatedDetailsToVideoCodecConfigs(
       const JsepTrackNegotiatedDetails& aDetails,
-      std::vector<UniquePtr<VideoCodecConfig>>* aConfigs);
+      std::vector<VideoCodecConfig>* aConfigs);
 
   static void UpdateConduitRtpExtmap(
       MediaSessionConduit& aConduit, const JsepTrackNegotiatedDetails& aDetails,